### PR TITLE
Added travis checks for missing docs and tests

### PR DESCRIPTION
**Pull request by garethahealy**
#### What is this PR About?
Docs and tests should be added for all new pipeline-lib methods. If they are not included, fail the build.

#### How do we test this?
Run travis job with missing files, see it fail. Add the files, see it pass.

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,15 @@ install:
   - mvn -N io.takari:maven:wrapper -Dmaven=3.6.0
   - java -version && javac -version && ./mvnw -version -B
 
+before_script:
+  # Check each method has docs
+  - find vars -type f -name "*.groovy" -exec bash -c '[ -f vars/$(basename {} .groovy).txt ] || echo "Missing doc: - vars/$(basename {} .groovy).txt"' \;
+  - find vars -type f -name "*.groovy" -exec bash -c '[ -f vars/$(basename {} .groovy).txt ] || exit 1' \;
```
Comment (pabrahamsson) [inline .travis.yml]: If I'm reading this right we should have `$?` be `1` if we're missing docs but I'm not seeing that. Are you able to confirm?
```
$ find vars -type f -name "*.groovy" -exec bash -c '[ -f vars/$(basename {} .groovy).txt ] || echo "Missing doc: - vars/$(basename {} .groovy).txt"' \;
$ find vars -type f -name "*.groovy" -exec bash -c '[ -f vars/$(basename {} .groovy).txt ] || exit 1' \;
$ echo $?
0
$ touch vars/test.groovy
$ find vars -type f -name "*.groovy" -exec bash -c '[ -f vars/$(basename {} .groovy).txt ] || echo "Missing doc: - vars/$(basename {} .groovy).txt"' \;
Missing doc: - vars/test.txt
$ find vars -type f -name "*.groovy" -exec bash -c '[ -f vars/$(basename {} .groovy).txt ] || exit 1' \;
$ echo $?                                                                                               
0
```

File: .travis.yml [hunk 2]
```diff
@@ -7,6 +7,15 @@ install:
...
+
+  # Check each method has a test
+  - find vars -type f -name "*.groovy" -exec bash -c '[ -f test/Jenkinsfile-$(basename {} .groovy) ] || echo "Missing test: - test/Jenkinsfile-$(basename {} .groovy)"' \;
+  - find vars -type f -name "*.groovy" -exec bash -c '[ -f test/Jenkinsfile-$(basename {} .groovy) ] || exit 1' \;
```
Comment (pabrahamsson) [inline .travis.yml]: Same as above

File: .travis.yml (hunk 1 above)
Comment (garethahealy) [reply .travis.yml]: Simplified/corrected check

**Comment by garethahealy** [comment]
dont merge/ignore - bad push.

**Comment by garethahealy** [comment]
@pabrahamsson ; pushed changes, ready for review again.

File: .travis.yml [hunk 3]
```diff
@@ -7,6 +7,12 @@ install:
   - mvn -N io.takari:maven:wrapper -Dmaven=3.6.0
   - java -version && javac -version && ./mvnw -version -B
 
+before_script:
+  # Check each method has docs and tests, if missing, fail
+  - find vars -type f -name "*.groovy" -exec bash -c '[ -f vars/$(basename {} .groovy).txt ] || echo "Missing vars/$(basename {} .groovy).txt" >> fail.tmp' \;
+  - find vars -type f -name "*.groovy" -exec bash -c '[ -f test/Jenkinsfile-$(basename {} .groovy) ] || echo "Missing test/Jenkinsfile-$(basename {} .groovy)" >> fail.tmp' \;
+  - [ -f fail.tmp ] && cat fail.tmp && exit 1
```
Comment (pabrahamsson) [inline .travis.yml]: Will this exit the shell in the case of missing docs or tests? Might be ok as I'm assuming that Travis will just exit the current run and report a failure.

**Pull request merged**